worlds/pokemon_crystal/locations.py

```diff
@@ -0,0 +1,75 @@
+from typing import TYPE_CHECKING, Optional, Dict, FrozenSet
+
+from BaseClasses import Location, Region
+from .data import data
+
+if TYPE_CHECKING:
+    from . import PokemonCrystalWorld
+
+
+class PokemonCrystalLocation(Location):
+    game: str = "Pokemon Crystal"
+    rom_address: Optional[int]
+    default_item_code: Optional[int]
+    flag: Optional[int]
+    tags: FrozenSet[str]
+
+    def __init__(
+            self,
+            player: int,
+            name: str,
+            parent: Optional[Region] = None,
+            flag: Optional[int] = None,
+            rom_address: Optional[int] = None,
+            default_item_value: Optional[int] = None,
+            tags: FrozenSet[str] = frozenset()
+    ) -> None:
+        super().__init__(player, name, flag, parent)
+        self.default_item_code = default_item_value
+        self.rom_address = rom_address
+        self.tags = tags
+
+
+def create_locations(world: "PokemonCrystalWorld", regions: Dict[str, Region]) -> None:
+    exclude = set()
+    if not world.options.randomize_hidden_items:
+        exclude.add("Hidden")
+    if not world.options.randomize_pokegear:
+        exclude.add("Pokegear")
+    if not world.options.trainersanity:
+        exclude.add("Trainersanity")
+    if not world.options.randomize_badges:
+        exclude.add("Badge")
+    if not world.options.randomize_berry_trees:
+        exclude.add("BerryTree")
```

Comment:
Since I know this was based on Emerald, I feel I should point out that I've changed it in #3263 so that category is used instead of tags now, which feels more correct to me since these are implicitly mutually exclusive. And it looks like you have a `type` field already. I'd suggest checking that PR to see if you want to make similar changes. This system works fine enough as it is though with the way these tags are treated in actuality.

Reply:
The way this ended up working for me is `type` is used for the ROM side, `tags` for the logic. The machine part in Cerulean gym is a `hiddenitem` for patching but not in logic. I could move the logic classifications to their own attribute but the tags would still stick around for other things.

Reply:
Ah, so `type` changes how you patch the item in. That's not something that matters in Emerald.

Up to you. Just wanted to make sure it had thought put into it and wasn't just duplicated from Emerald since I've abandoned it in this form.